app/controllers/close_contacts_controller.rb

```diff
@@ -42,4 +42,15 @@ def update
                                created_by: current_user.email,
                                comment: "User edited a close contact (ID: #{cc.id}).")
   end
+
+  # Delete an existing close contact record
+  def destroy
+    redirect_to root_url unless current_user.can_create_patient_close_contacts?
```

Comment:
this ensures that there won't be multiple renders

```suggestion
    redirect_to(root_url) && return unless current_user.can_create_patient_close_contacts?
```